<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Mapcount by mpcox</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>Mapcount</h1>
        <p>Counts read hits against a reference gene set</p>

        <p class="view"><a href="https://github.com/mpcox/mapcount">View the Project on GitHub <small>mpcox/mapcount</small></a></p>


        <ul>
          <li><a href="https://github.com/mpcox/mapcount/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/mpcox/mapcount/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/mpcox/mapcount">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
        <h1>
<a id="mapcount" class="anchor" href="#mapcount" aria-hidden="true"><span class="octicon octicon-link"></span></a>mapcount</h1>

<p>The Perl program <strong>mapcount</strong> counts next generation sequencing reads that map against a reference gene set.</p>

<p><strong>mapcount</strong> extracts a list of expected sequence IDs from a user-provided FASTA file (such as a file of gene models).  It then parses a user-provided mapping file in the standard SAM format and counts the number of reads that map to each gene.  Tabular output is written to a user-defined text file.</p>

<p>Optional flags allow the user to:</p>

<ul>
<li>Trim the sequence ID before a space (<em>e.g.</em>, "comp18530_c0_seq1 len=1406 path=[683:0-1405]" would instead be represented as "comp18530_c0_seq1").</li>
<li>Define the minimum match length to include the read (<em>e.g.</em>, all matching reads smaller than some specified value, such as 25 base pairs, can be excluded from the output table).</li>
<li>Count paired end reads as only a single match. (Although there might be two paired reads, these represent only one single underlying nucleotide fragment).</li>
</ul>

<p>Program usage is as follows:</p>

<pre><code>map_count -f|fasta FASTA file -s|sam SAM file -o|output text file 
         [-t|trim] [-l|length 25] [-p|pair]
</code></pre>

<pre><code>Required flags:
-f|fasta    FASTA reference file
-s|sam      SAM mapping file
-o|output   tabular text file
</code></pre>

<pre><code>Optional flags:
-t|trim     trim ID to entry before first space
-l|length   minimum match length for inclusion in the count table
-p|pair     count paired matches as one match
</code></pre>
      </section>
      <footer>
        <p>This project is maintained by <a href="https://github.com/mpcox">mpcox</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
    
  </body>
</html>